### PR TITLE
Fixed mapInPlace on mutable maps and sets

### PR DESCRIPTION
**Pull request by ruhly**
addresses #408

### CODE REVIEW

**Comment by ruhly** [comment]
Once the mapInPlace issue is resolved, it probably needs to be propagated to flatMapInPlace because it looks like those methods suffer from the same issue

File: collections/src/main/scala/strawman/collection/mutable/Map.scala [hunk 1]
```diff
@@ -121,15 +121,10 @@ trait MapOps[K, V, +CC[X, Y] <: MapOps[X, Y, CC, _], +C <: MapOps[K, V, CC, C]]
...
-      val mapped = f(elem)
-      if (!contains(mapped._1)) {
-        toAdd += mapped
-        toRemove -= elem._1
-      }
+        toAdd += f(elem)
     }
-    for (elem <- toRemove) coll -= elem
+    coll.clear()
     for (elem <- toAdd) coll += elem
```
Comment (julienrf) [inline collections/src/main/scala/strawman/collection/mutable/Map.scala]: You can even call `coll ++= toAdd`, which should be faster.

**Comment by szeiger** [comment]
I didn't even realize we had these methods. If any such method makes sense for `Map` then IMHO it is `mapValuesInPlace`.

**Comment by ruhly** [comment]
also see issue #473.

**Comment by julienrf** [comment]
Thanks

**Pull request merged**